packages/commonwealth/client/scripts/helpers/ContractHelpers/NamespaceFactory.ts

```diff
@@ -152,6 +152,44 @@ class NamespaceFactory extends ContractBase {
     }
     return txReceipt;
   }
+
+  async newContest(
+    namespaceName: string,
+    contestInterval: number,
+    winnerShares: number[],
+    stakeId: number = 2,
+    prizeShare: number,
+    voterShare: number,
+    feeShare: number,
+    weight: number = 1,
+    walletAddress: string,
+  ): Promise<any> {
+    if (!this.initialized || !this.walletEnabled) {
+      await this.initialize(true);
+    }
+    let txReceipt;
+    try {
+      txReceipt = await this.contract.methods
+        .newContest(
+          namespaceName,
+          contestInterval,
+          winnerShares,
+          stakeId,
+          prizeShare,
+          voterShare,
+          feeShare,
+          weight,
+        )
+        .send({
+          from: walletAddress,
+          maxPriorityFeePerGas: null,
+          maxFeePerGas: null,
+        });
+    } catch {
+      throw new Error('Transaction failed');
+    }
+    return txReceipt;
```

Comment:
I think I would need contest address to do followup API call to the backend. Is it somewhere inside `txReceipt` ?

Reply:
realizing the pattern might have caused some confusion without explanation: 

You'll want to call Contest.ts:NewRecurringContest which internally calls this. The function on the contest helper you'll call will abstract away contract address lookup and return you a contract address directly

Reply:
on the UI I am able to pick if the contest is recurring or not


<img width="707" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/e6a68264-985a-4c8d-a7f0-7f85bdf5822a">

My understanding was that depending on chosen option, I call either `newRecurringContest` or `newContest`

Reply:
Yes that is correct Yes = newRecurring, No = newSingle